api_app/api/routes/costs.py

```diff
@@ -83,3 +107,20 @@ async def workspace_costs(workspace_id: UUID4, params: CostsQueryParams = Depend
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=strings.WORKSPACE_DOES_NOT_EXIST)
     except SubscriptionNotSupported:
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=strings.API_GET_COSTS_SUBSCRIPTION_NOT_SUPPORTED)
+    except TooManyRequests as e:
+        return JSONResponse(content={
+                            "error": {
+                                "code": "429",
+                                "message": strings.API_GET_COSTS_TOO_MANY_REQUESTS,
+                                "retry-after": str(e.retry_after)
+                            }}, status_code=429, headers={"Retry-After": str(e.retry_after)})
+    except ServiceUnavailable as e:
+        return JSONResponse(content={
+                            "error": {
+                                "code": "503",
+                                "message": strings.API_GET_COSTS_SERVICE_UNAVAILABLE,
+                                "retry-after": str(e.retry_after)
+                            }}, status_code=503, headers={"Retry-After": str(e.retry_after)})
+    except Exception as e:
+        logging.error("Failed to query Azure TRE workspace costs", exc_info=e)
+        raise e
```

Comment:
Why raise and not return 500?

Reply:
you are correct, on the router I should have just return a 500 and not rethrow the exception.
I'll fix that. thanks